scripts/gee

```diff
@@ -1807,20 +1816,41 @@ function _update_from_upstream() {
   local LOCAL_BRANCH="$1"
   local UPSTREAM_BRANCH="$2"
   _checkout_or_die "${LOCAL_BRANCH}"
-  if ! _git pull --rebase --autostash upstream "${UPSTREAM_BRANCH}"; then
-    _warn "Git pull operation failed."
-    if ! _is_rebase_in_progress; then
-      _fatal "Rebase not in progress, no idea what went wrong."
+  # I don't know why, but sometimes github rejects pull requests right after
+  # merging PRs.
+  #    $ /usr/bin/git pull --rebase --autostash upstream master
+  #    ssh_dispatch_run_fatal: Connection to 192.30.255.113 port 22: Broken pipe
+  #    fatal: Could not read from remote repository.
+  #
+  #    Please make sure you have the correct access rights
+  #    and the repository exists.
+  #
+  # The command always suceeds on retry.  So -- let's just enable retries for
+  # this operation.
+  local TRIES=0
+  local MAX_RETRIES=3
+  while [[ "$TRIES" -lt "$MAX_RETRIES" ]]; do
+    if _git pull --rebase --autostash upstream "${UPSTREAM_BRANCH}"; then
+      # success
+      break
     fi
-    _interactive_conflict_resolution "upstream/${UPSTREAM_BRANCH}" "${LOCAL_BRANCH}"
+    _warn "Git pull operation failed."
     if _is_rebase_in_progress; then
-      local STATUS
-      mapfile -t STATUS < <( "${GIT}" status );
-      _warn "${STATUS[@]}"
-      _warn "Merge conflict in branch ${LOCAL_BRANCH}, must be manually resolved."
-      _fatal "Exited without resolving rebase conflict."
+      _interactive_conflict_resolution "upstream/${UPSTREAM_BRANCH}" "${LOCAL_BRANCH}"
```

Comment:
It's not enough just to retry the git pull? Do you have to do also `_interactive_conflict_resolution` after every try?

Reply:
It won't do `_interactive_conflict_resolution` unless `_is_rebase_in_progress` is true.  That function checks to see if the git branch is in a "awaiting conflict resolution during a rebase operation" state.  If the pull operation just fails the way I've been seeing, this function will return false.

There isn't a way to distinguish between the error codes of `gh pull` (or at least it isn't documented), but this check gives me a reliable indicator of what needs to be done.